core/src/main/web/app/mainapp/components/notebook/cell.jst.html

```diff
@@ -45,6 +45,7 @@
     </div>
     <div class="cell-menu-item evaluate" ng-if="isMarkdownCell()" title="run cell"></div>
     <div class="cell-menu-item evaluate" ng-click="evaluate($event)" ng-if="isCodeCell()" title="run cell"></div>
+    <div class="cell-menu-item evaluate" ng-if="isSection()" ng-click="evaluateSection($event)" title="run all cells"></div>
```

Comment:
As I said above you can ng-if="cellmodel.type === 'section'" - no need for the isSection function, we already have the answer and it won't change, we don't need to compute it all the time.

Also, since the type of a section won't change, we don't need to watch it all the time, so we can use one-time binding, like this:

ng-if="::cellmodel.type === 'section'" - the double colon at the start of the expression marks it as one time evaluation. The watcher will run once when the section is created and never again. This helps us reduce the number of watchers that fire on every digest.